### PR TITLE
[core services] added core services that will be used all over the ap…

### LINKED ISSUE #13: Setup core services
Setup core services like 
1. Dialog Service
1. Loader service
1. Notification Service
1. Error handler service

### PR DESCRIPTION
**Pull request by gastonche**
…p, including, dialogs, notifications, loaders and generic error handlers

## Description
This PR includes changes pertaining to core services to be used in the application. 
Services include:
1. Modal Service
1. Notification Service
1. Loader Service
1. Http Error handler

Fixes #(issue number 1) #(issue number 2 if applicable)

## How Has This Been Tested?
I have integrated storybook which can be used as a sandbox to test new UIs and functionalities. To test this please run ```yarn install``` and then ```yarn run storybook```

Then go the the tab for services and see the various services that have been created. These are generic services, so almost everything can be customized by the person using them

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by sonegillis — COMMENTED** [review]
Really great work there @gastonche. I have gone through the code added on this commit and I must say it was much effort put in there. But I have something I will like to highlight.
 
These are services to be used throughout the app by other developer(s), yet it doesn't have any commenting explicitly detailing how to use the services you created. Lots of functions with arguments and it's difficult to know how the argument could be used by the caller of the function or how the function uses the argument. Since these are services that are going to be generally used, I see it necessary for a properly commented code or have a separate page where each of these services are used so that one can see example usage with code. As an example,
```javascript
export default function Loader({
  hide, cancel, fn, message
}) {
  fn().then(hide).catch(cancel);
  return (<div className="loader-dialog">
    <img src={IMG} alt="loading..."/>
    { message }
  </div>);
}
```
You might tell us in a comment what `hide`, `cancel` and `fn` functions should do or create a usage page (for local test purpose) where the loader has been called and used

Also as a suggestion, rather than creating services or components from scratch which have been developed in very good libraries out there (bootstrap, prime react, bulma and many others) with good documentation, why not install these libraries and make use of their components to speed up development.

Lastly I think installing prettier js will help us to have nicely formatted and clean code that can be easily readable

**Comment by gastonche** [comment]
> Really great work there @gastonche. I have gone through the code added on this commit and I must say it was much effort put in there. But I have something I will like to highlight.
> 
> These are services to be used throughout the app by other developer(s), yet it doesn't have any commenting explicitly detailing how to use the services you created. Lots of functions with arguments and it's difficult to know how the argument could be used by the caller of the function or how the function uses the argument. Since these are services that are going to be generally used, I see it necessary for a properly commented code or have a separate page where each of these services are used so that one can see example usage with code. As an example,
> 
> ```js
> export default function Loader({
>   hide, cancel, fn, message
> }) {
>   fn().then(hide).catch(cancel);
>   return (<div className="loader-dialog">
>     loading...
>     { message }
>   </div>);
> }
> ```
> 
> You might tell us in a comment what `hide`, `cancel` and `fn` functions should do or create a usage page (for local test purpose) where the loader has been called and used
> 
> Also as a suggestion, rather than creating services or components from scratch which have been developed in very good libraries out there (bootstrap, prime react, bulma and many others) with good documentation, why not install these libraries and make use of their components to speed up development.
> 
> Lastly I think installing prettier js will help us to have nicely formatted and clean code that can be easily readable

1. In the description of the pull request added I added commands to be able to test these services in storybook. But thanks for pointing out the commenting issue.

1. As for writing the services from scratch, my issue comes with the way these libraries have implemented their components, for example, you need to display a dialog in a page and keep a flag in order to show it or not show it, you need to mix the dialog logic and the page logix to get it to work properly, but in my experience that is harder and more error prone, that Is why all these services are implemented with a more promise based approach, fire the service, wait for it to resolve(hide) or reject(cancel) and know what to do in either case.

But Thanks @sonegillis I'll try to comment each service more. You can checkout the storybooks for usage tho.

File: stories/0-Welcome.stories.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (wadleo) [inline stories/0-Welcome.stories.js:1]: @gastonche I guess we will have to, in the long run, remove this sample files right? (that's if I'm getting the jist of this package well)

File: src/core/services/dialog/Dialog.css [hunk 2]
```diff
@@ -0,0 +1,125 @@
+.dialog-wrapper {
```
Comment (wadleo) [inline src/core/services/dialog/Dialog.css:1]: these services look cool

**Review by wadleo — COMMENTED** [review]
I feel we should not commit them, engineers should check the samples/docs online, what do you guess think? Let the frontend team members decide.

File: stories/0-Welcome.stories.js (hunk 1 above)
Comment (wadleo) [reply stories/0-Welcome.stories.js:1]: What I vetoing for here is exactly opposite of @sonegillis review(I am just reading the other comments from reviews here now).
 
@sonegillis I think @gastonche already include a whole [module](https://github.com/BuildForSDG/Team-226-Frontend/tree/ft-build-core-services/stories) of samples in the project with directions in this issue description on how to access them on the browser and see the code, that's awesome. 

But I think developers should look at those examples online or they should create a separate branch for this module of samples and not merged into the develop branch IMHO.

Comment (wadleo) [reply stories/0-Welcome.stories.js:1]: Apart from this, the frontend guys are good to go with this. For me, let them vote on how to integrate the changes and move ahead.

**Review by sonegillis — APPROVED** [review]

File: stories/0-Welcome.stories.js (hunk 1 above)
Comment (wadleo) [reply stories/0-Welcome.stories.js:1]: I will close this for now, great work @gastonche

**Review by wadleo — APPROVED** [review]
Nice work

**Review by Cressence — APPROVED** [review]
Services look good. Nice @gastonche

**Pull request merged**